be/src/storage/persistent_index.cpp

```diff
@@ -863,6 +866,7 @@ class FixedMutableIndex : public MutableIndex {
     }
 
     Status insert(const Slice* keys, const IndexValue* values, const std::vector<size_t>& idxes) override {
+        CHECK_MEM_LIMIT("FixedMutableIndex::insert");
```

Comment:
Why not use `TRY_CATCH_BAD_ALLOC` ?

Reply:
> Why not use `TRY_CATCH_BAD_ALLOC` ?

There is no concurrency issue here.

Reply:
But it can still oom when do insert or upsert.

Reply:
E,g, when you call `CHECK_MEM_LIMIT("FixedMutableIndex::insert");` , it's not OOM yet, but after them, you keep insert kvs into pk index, then it will be OOM.

Reply:
> E,g, when you call `CHECK_MEM_LIMIT("FixedMutableIndex::insert");` , it's not OOM yet, but after them, you keep insert kvs into pk index, then it will be OOM.

done.